astropy/io/fits/tests/test_connect.py

```diff
@@ -155,6 +155,14 @@ def test_read_with_nonstandard_units(self):
         t = Table.read(hdu)
         assert equal_data(t, self.data)
 
+    def test_read_include_columns(self):
+        include_columns = ['a', 'c']
+        hdu = BinTableHDU(self.data)
+
+        t = Table.read(hdu, include_columns=include_columns)
+        assert len(t.columns) == 2
```

Comment:
If order is preserved then it seems like `assert t.colnames == include_columns` would be sufficient?